### PR TITLE
RELATED: ONE-4647 Custom styles for Widget in KD

### PR DESCRIPTION
**Pull request by NestorEncinas**
Custom styles for Widget in KD

- Add custom kpiDashboard keys on Theme Loader 
        -  borderWidth, borderRadius, dropShadow

- Remove derived kpiDashboard color 

JIRA: ONE-4647 

Update kpiDashboards metadata theme object

- Rename MD theme object kpiDashboard to `title`

JIRA: ONE-4647

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1362/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2950/ : SUCCESS in 13m 25s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1372/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2961/ : SUCCESS in 13m 17s

File: libs/sdk-ui-theme-provider/src/cssProperties.ts [hunk 1]
```diff
@@ -48,6 +47,15 @@ const customParserFunctions: ParserFunction[] = [
     { key: "--gd-button-borderRadius", fn: (value: string) => `${value}px` },
     { key: "--gd-button-textCapitalization", fn: (value: boolean) => (value ? "capitalize" : undefined) },
     { key: "--gd-button-dropShadow", fn: (value: boolean) => (value ? undefined : "none") },
+    {
+        key: "--gd-kpiDashboards-content-widget-borderWidth",
+        fn: (value: number) => (value ? `${value}px` : "0px"),
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-theme-provider/src/cssProperties.ts:52]: do we want to handle default value here if we already have it in 
$gd-kpiDashboards-content-widget-borderWidth: var(
    --gd-kpiDashboards-content-widget-borderWidth,
    $item-border
);
in KD?

File: libs/sdk-ui-theme-provider/src/cssProperties.ts [hunk 2]
```diff
@@ -5,7 +5,6 @@ import { IThemePalette, ITheme } from "@gooddata/sdk-backend-spi";
 
 // keep it in sync with SCSS:$gd-color-text-light
 const GD_COLOR_TEXT_LIGHT = "#fff";
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-theme-provider/src/cssProperties.ts:7]: both commits should have commit msg according contribution doc:
"Also note how to add JIRA ticket reference: using a "JIRA: [ticket-id]" to the end of the commit body"

**Review by ivan-nejezchleb — CHANGES_REQUESTED** [review]

File: libs/sdk-ui-theme-provider/src/cssProperties.ts (hunk 1 above)
Comment (NestorEncinas) [reply libs/sdk-ui-theme-provider/src/cssProperties.ts:52]: If the user doesn't want to have border-width?

Comment (ivan-nejezchleb) [reply libs/sdk-ui-theme-provider/src/cssProperties.ts:52]: then he needs to put there explicitly zero and check in function should be value !== undefined to accept 0

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1381/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2971/ : SUCCESS in 14m 01s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1383/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2974/ : SUCCESS in 13m 45s

**Review by ivan-nejezchleb — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/477/ : SUCCESS in 35s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/736/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/2977/ : SUCCESS in 20m 09s

**Pull request merged**